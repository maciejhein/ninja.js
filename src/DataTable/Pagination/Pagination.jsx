import React from 'react'

import PaginationItems from './PaginationItems'

const Pagination = ({ currentPageNumber, totalNumberOfPages, onChange }) => {
  if(totalNumberOfPages <= 1){
    return null;
  }

  const pages =Array.from(Array(totalNumberOfPages).keys());

  return(
    <ul className="pagination">
      {
        pages.map(pageNumber => {
          return <PaginationItems
            key={pageNumber}
            isActive={currentPageNumber === pageNumber}
            currentPageNumber={currentPageNumber}
            pageNumber={pageNumber}
            onChange={onChange} />
        })
      }
    </ul>
  )
}

export default Pagination
