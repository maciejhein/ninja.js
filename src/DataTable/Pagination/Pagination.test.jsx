import React from 'react';
import { mount, shallow } from 'enzyme';

import Pagination from './Pagination';

jest.mock('./PaginationItems', () => () => 'PaginationItems');

describe('Pagination', () => {
  let props; 

  beforeEach(() => {
    props = {
      currentPageNumber: 1,
      totalNumberOfPages: 3,
      onChange: jest.fn()
    };
  });

  it('should renders correctly', () => {
    const component = shallow(<Pagination {...props}  />);

    expect(component).toMatchSnapshot();
  });

  it('should returns empty object when `totalNumberOfPages` is less than one', () => {
    const newProps = {
      ...props,
      totalNumberOfPages: 0
    }

    const component = mount(<Pagination {...newProps} />);

    expect(component).toEqual({}); 
  });
});