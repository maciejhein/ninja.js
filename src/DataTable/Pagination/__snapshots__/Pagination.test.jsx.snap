// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should renders correctly  1`] = `
<ul
  className="pagination"
>
  <Component
    currentPageNumber={1}
    isActive={false}
    key="0"
    onChange={[MockFunction]}
    pageNumber={0}
  />
  <Component
    currentPageNumber={1}
    isActive={true}
    key="1"
    onChange={[MockFunction]}
    pageNumber={1}
  />
  <Component
    currentPageNumber={1}
    isActive={false}
    key="2"
    onChange={[MockFunction]}
    pageNumber={2}
  />
</ul>
`;
